src/plot_settings.py

```diff
@@ -0,0 +1,177 @@
+"""
+plot_settings.py
+================
+
+usage:
+
+from src.plot_settings import ps_defaults, label_subplots, get_dim, set_dim, PALETTE
+
+ps_defaults(use_tex=True)
+
+# ---- example set of graphs ---
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+fig, axs = plt.subplots(2, 2)
+x=np.linspace(0, np.pi, num=100)
+axs[0, 0].plot(x, np.sin(x)); axs[0, 1].plot(x, np.cos(x))
+axs[1, 0].plot(x, np.sinc(x)); axs[1, 1].plot(x, np.abs(x))
+
+# set size
+set_dim(fraction_of_line_width=1, ratio=(5**.5 - 1) / 2)
+
+# label subplots
+label_subplots(axs, start_from=0, fontsize=10)
+
+"""
+import numpy as np
+import itertools
+from distutils.spawn import find_executable
+import matplotlib
+import matplotlib.style
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+
+def ps_defaults(use_tex=True):
+    """
+    Apply plotting style to produce nice looking figures.
+    Call this at the start of a script which uses matplotlib,
+    and choose the either high (uses latex) or low (without latex)
+    """
+    #'#CB4154' ,
+    # matplotlib.use('agg') this used to be required for jasmin
+    p_general = {
+        "font.family": "serif",
+        "font.serif": [],
+        # Use 10pt font in plots, to match 10pt font in document
+        "axes.labelsize": 10,
+        "font.size": 10,
+        # Make the legend/label fonts a little smaller
+        "legend.fontsize": 8,
+        "xtick.labelsize": 8,
+        "ytick.labelsize": 8,
+        # Set the font
+        "mathtext.fontset": "cm",
+        # "font.family": "STIXGeneral",
+        "font.sans-serif": ["DejaVu Sans"],  # gets rid of error messages
+        "font.monospace": [],
+        "lines.linewidth": 0.75,
+        "scatter.marker": "+",
+        "image.cmap": "RdYlBu_r",
+    }
+    matplotlib.rcParams.update(p_general)
+    matplotlib.style.use("seaborn-colorblind")
+
+    if use_tex and find_executable("latex"):
+        p_setting = {
+            "pgf.texsystem": "pdflatex",
+            "text.usetex": True,
+            "pgf.preamble": (r"\usepackage[utf8x]{inputenc} \usepackage[T1]{fontenc}"
+                             +r"\usepackage[separate -uncertainty=true]{siunitx}"),
+        }
+    else:
+        p_setting = {
+            "text.usetex": False,
+        }
+    matplotlib.rcParams.update(p_setting)
+
+
+def label_subplots(axs, start_from=0, fontsize=10):
+    """
+    Currently adds (a), (b), (c) etc. at the top left of each subplot panel.
+    Labelling order achieved through ravelling the input list/ array
+    :param axs: list or array of subplot axes.
+    :param start_from: lets you start from a different axes.
+    :param fontsize: in points
+    :return: void; alters axes objects
```

Comment:
nit:  This is not really necessary, but the rest of the project uses [Google style formatted docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) (slightly more readable). If we wanted to have an automatic documentation after by [running Sphinx](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html) over the project we'd need to harmonize the docstring formats

Reply:
wilco